src/Umbraco.Core/ObjectExtensions.cs

```diff
@@ -156,6 +156,11 @@ public static Attempt<object> TryConvertTo(this object input, Type target)
                         if (inner.Success)
                         {
                             input = inner.Result; // Now fall on through...
+

```

Comment:
Ah we can't really be that specific. Going to have the same issue with ints, anything really. What happens is... if `target` is nullable, then we already convert `input` to the nullable underlying type, and change `input` with that value. So... right before the line near the end that does `var convertible2 = ...`, you need to handle the special case, with something along

```
if (target.IsGenericType && GetCachedGenericNullableType(target) != null)
{
  // cannot Convert.ChangeType as that does not work with nullable
  // input has already been converted to the underlying type - just
  // return input, there's an implicit conversion from T to T? anyways
  return Attempt.Succeed(input);
}
```

so basically the only thing we change is ... if `target` is a nullable there's no need to go through the `IConvertible` thing, just return `input` should be OK.

Making sense?

Reply:
Owww, yes, much simple and elegant. Will take care of it first thing in the morning.

Reply:
I've made this changed and pushed it. I've moved all the TryConvertTo tests to the ObjectExtensionsTests where the others live.

All tests pass now, but I've also added a TODO note if @zpqrtbnk can have a look? We do a check for input of `string` and then check `IsNullOrEmpty` and then specifically check for `datetime` and `bool` and return null but that seems odd right? Why the bool/datetime check, if the string IsNullOrEmpty and the destintation isn't a string, but its nullable, wouldn't it always just be null ?